### PR TITLE
Added length restriction on log output

### LINKED ISSUE #13: Add length restriction on log output
- [x] Limit the length of each log message (including header i.e., prefix of message)
- [x] Add return type int to `log_this` function
- [x] return >0 for Success: Message logged and Positive returned value is the length of logged message
- [x] return =0 for Failure: Message ignored because of lower-priority
- [x] return <0 for Failure: Overflow occurred. Negative returned value is the required length to log input log-request
- [x] parse error codes in LOG macro and write overflow text when overflow occurs
- [x] Add snippet for overflow i.e., logging a very lengthy message to watch overflow
- [x] Add test for overflow using log_this function directly

### PR DESCRIPTION
**Pull request by khushal-banks**
This is PR to fix #13 

### Related Notes (for later)
Currently we are only logging on stderr. But later after integration of log-files we would need to add following. Following are delayed just to make sure how writing streams would work differently on windows.

- [ ] Consider flushing write-stream
- [ ] Consider lock/unlock of write-stream

### CODE REVIEW

File: tests/test04_overflow.c [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    locate.file = FILENAME;
+    locate.func = __func__;
+    locate.line = 0;
+
+    int status = log_this(
+        LOG_FATAL, locate, "%s%s%s%s%s%s%s%s%s%s", "1234567890ABCDEFGHIJ",
+        "1234567890ABCDEFGHIJ", "1234567890ABCDEFGHIJ", "1234567890ABCDEFGHIJ",
+        "1234567890ABCDEFGHIJ", "1234567890ABCDEFGHIJ", "1234567890ABCDEFGHIJ",
+        "1234567890ABCDEFGHIJ", "1234567890ABCDEFGHIJ", "1234567890ABCDEFGHIJ");
+
```
Comment (khushal-banks) [inline tests/test04_overflow.c:15]: - Add comments
- Consider if status variable needs a different name

File: logbank/include/logbank/logbank.h [hunk 2]
```diff
@@ -133,7 +139,15 @@ typedef struct {
             locate.line = __LINE__;                                                      \
         }                                                                                \
                                                                                          \
-        log_this(level, locate, __VA_ARGS__);                                            \
+        int status = log_this(level, locate, __VA_ARGS__);                               \
+        if (status < 0) {                                                                \
+                                                                                         \
```
Comment (khushal-banks) [inline logbank/include/logbank/logbank.h:144]: Consider renaming status variable.

Move this empty line to before next log_this() call to make it a bit more readable.

File: logbank/src/logbank.c [hunk 3]
```diff
@@ -176,6 +176,104 @@ log_stamp_collector(char* stamp, const char* format, size_t size) {
...
+
+/*!
+ * \fn      log_stream_write(const char* fmt, ...)
+ * \brief   write directly to configured stream
+ *
+ * \param[in]   message      pointer to log-message buffer to write
+ */
+static void
+log_stream_write(const char* message) {
+    (void)fprintf(stderr, "%s", message);
```
Comment (khushal-banks) [inline logbank/src/logbank.c:274]: Consider using flush on stream and maybe locking/unlocking the stream

File: logbank/src/logbank.c [hunk 4]
```diff
@@ -176,6 +176,104 @@ log_stamp_collector(char* stamp, const char* format, size_t size) {
...
+    }
+
+    /* log separator between header (message-prefix) and message */
+    len += snprintf(header + len, size - len, ": ");
+
+    /* terminate the use of colors */
+    len += LOG_USE_COLOR
+               ? snprintf(header + len, size - len, "%s", color_list[LOG_COLOR_END])
+               : 0;
+
```
Comment (khushal-banks) [inline logbank/src/logbank.c:262]: There are simply too many snprintf calls with almost same arguments
Consider adding a macro at least for how the colors are added

File: logbank/include/logbank/logbank.h (hunk 2 above)
Comment (khushal-banks) [reply logbank/include/logbank/logbank.h:144]: Status as a name is fine.
-ve length would be confusing.

Will add comments.

File: tests/test04_overflow.c (hunk 1 above)
Comment (khushal-banks) [reply tests/test04_overflow.c:15]: Status as a name is fine.
-ve length would be confusing.

File: logbank/src/logbank.c (hunk 3 above)
Comment (khushal-banks) [reply logbank/src/logbank.c:274]: It will be added after introduction to log-files.
